packages/svelte/src/internal/client/runtime.js

```diff
@@ -935,12 +959,49 @@ export function unsubscribe_on_destroy(stores) {
 	});
 }
 
+/**
+ * If the `current_derived_proxy_property` is not `null` then that means we should look at the current
+ * property on the object and see if actually need original derived object dependency. For example,
+ * if you had this:
+ *
+ * a.b.c
+ *
+ * Under-the-hood, `a` might be a derived signal, so we'd call get() on it. Resulting in `a` being a dependency
+ * for the currently active effect. The accessors to `b` and `c` would result in the `current_derived_proxy_property`
+ * changing to include ['b', 'c'] in the `current_derived_proxy_property.p` paths property. We can then use that to
+ * determine a new derived temporary signal that encapsulates a.b.c. This temporarly signal then becomes the dependency
+ * and we no longer need to the original depdency to `a` for the current effect. Thus making
+ */
+function capture_fine_grain_derived_property() {
+	const derived_property =
+		/** @type {{ s: import('./types.js').ComputationSignal, p: Array<string | symbol> }} */ (
+			current_derived_proxy_property
+		);
+	if (is_last_current_dependency(derived_property.s)) {
+		if (current_dependencies === null) {
+			current_dependencies_index--;
+		} else {
+			current_dependencies.pop();
+		}
+	}
+	const derived_prop = derived(() => {
+		let value = /** @type {any} */ (get(derived_property.s, true));
+		const property_path = derived_property.p;
+		for (let i = 0; i < property_path.length; i++) {
+			value = value?.[property_path[i]];
+		}
+		return value;
+	});
+	current_derived_proxy_property = null;
+	get(derived_prop, true);
+}
+
 /**
  * @template V
  * @param {import('./types.js').Signal<V>} signal
  * @returns {V}
  */
-export function get(signal) {
+export function get(signal, skip_derived_proxy = false) {
```

Comment:
Could you add `@param {boolean} [skip_derived_proxy] <brief explanation what this does>` to the function signature? (maybe defer to that comment below for a more deep explanation)